client/src/components/SignIn/SignInForm.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  FormControl, InputLabel,
+} from '@mui/material';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import OutlinedInput from '@mui/material/OutlinedInput';
+import IconButton from '@mui/material/IconButton';
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import * as React from 'react';
+import IAuth from '../../Interfaces/auth';
+import { SignInValid } from '../../Validation';
+
+const SignIn = () => {
+  const [userData, setUserData] = React.useState<IAuth>({
+    showPassword: false,
+    password: '',
+    email: '',
+  });
+  const handleState = (e:React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.currentTarget;
+    setUserData((prev:object) => ({ ...prev, [name]: value }));
+  };
+  const handleClickShowPassword = (e:React.MouseEvent<HTMLButtonElement>) => {
+    if (e.currentTarget.name === 'password') {
+      setUserData({
+        ...userData,
+        showPassword: !userData.showPassword,
+      });
+    } else {
+      setUserData({
+        ...userData,
+        showConfirmPassword: !userData.showConfirmPassword,
+      });
+    }
+  };
+  const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  return (
+    <form onSubmit={(e) => {
+      e.preventDefault();
+      SignInValid.validate(userData)
+        .then(() => toast.success('WellCome !'))
+        .catch((err:any) => {
+          toast.error(err.message);
+        });
+    }}
+    >
+      <FormControl>
```

Comment:
You can create a input component and use it more than once in more than one place and pass it on what properties it needs, instead of repeating it and making it repeatedly.